flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/SingleInputGate.java

```diff
@@ -616,6 +628,13 @@ public void sendTaskEvent(TaskEvent event) throws IOException {
 		}
 	}
 
+	@Override
+	public void onExactlyOnceCheckpointCompletedOrCanceled(long checkpointId) {
+		for (InputChannel inputChannel: inputChannels.values()) {
```

Comment:
`inputChannels` are accessed via synchronized `requestLock` in other places. Do we need the sync here?

Reply:
I have removed this code